web3/contract.py

```diff
@@ -1290,34 +1282,41 @@ def createFilter(
         Create filter object that tracks logs emitted by this contract event.
         :param filter_params: other parameters to limit the events
         """
-        if fromBlock is None:
-            raise TypeError("Missing mandatory keyword argument to createFilter: fromBlock")
 
+        filter_builder = EventFilterBuilder(self._get_event_abi())
         if argument_filters is None:
             argument_filters = dict()
+        else:
+            for arg, arg_value in argument_filters.items():
+                filter_builder.args[arg].match_single(arg_value)
 
-        _filters = dict(**argument_filters)
+        if topics is not None:
+            raise NotImplementedError(
+                "Directly setting topics is not currently supported. "
+                "Use the argument_filters parameter instead.")
```

Comment:
I added this not implemented error to simplify things, as a temporary measure. The ability to set topics directly should be added back to createFilter

Reply:
I was hoping to merge the manually entered and the generated topic arguments together, but I realize that that would yield filter results that are entirely different than adding the results from the two sets of filtered data.  e.g. merging (A, B) with (C, D) to get a topic list like ((A, C), (B, D)).  The merged topic list would match AB, AD, CB, CD instead of just AC, BD.  For this reason, manually entering topics needs to either be disabled, or throw an exception if topics have already been generated.